manifests/server/config.pp

```diff
@@ -117,7 +117,7 @@
     }
 
   }
-  elsif ! $puppet::server_dynamic_environments {
+  elsif ! $puppet::server_dynamic_environments and !$puppet::server_directory_environments {
```

Comment:
For my reference: separate PR in https://github.com/theforeman/puppet-puppet/pull/286